test/solarwinds_apm/config_test.rb

```diff
@@ -0,0 +1,274 @@
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, :disabled, :enabled)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal :enabled
+    end
+
+    it 'with wrong env value and config value, use default :disabled' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, :disabled, :disabled)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal :disabled
+    end
+
+    it 'with wrong env value and wrong config value, use default :enabled' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'foo', :enabled)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal :enabled
+    end
+  end
+
+  describe '#enable_disable_config for boolean value' do
+    before do
+      SolarWindsAPM::Config.initialize
+    end
+
+    after do
+      ENV.delete('DUMMY_KEY')
+    end
+
+    # enable_disable_config(env_var, key, value, default, bool=false)
+    it 'with correct env true override config false ' do
+      ENV['DUMMY_KEY'] = 'true'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with correct env false override config false ' do
+      ENV['DUMMY_KEY'] = 'false'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong env, use default true ' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with wrong env, use default true ' do
+      ENV['DUMMY_KEY'] = 'foo'
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with config env true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, true, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with config env false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, false, true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default false' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', false, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal false
+    end
+
+    it 'with wrong config env, use default true' do
+      SolarWindsAPM::Config.enable_disable_config('DUMMY_KEY', :dummy_key, 'true', true, bool: true)
+      _(SolarWindsAPM::Config[:dummy_key]).must_equal true
+    end
+
+    it 'with nil env, config env true, use default false, should be true' do
```

Comment:
Minor, I think this test is only meaningful if the env var is set, i.e. `ENV['DUMMY_KEY'] = 'false'`

Reply:
Added.